bftengine/src/preprocessor/RequestProcessingState.cpp

```diff
@@ -56,9 +57,25 @@ void RequestProcessingState::handlePrimaryPreProcessed(const char *preProcessRes
       convertToArray(SHA3_256().digest(primaryPreProcessResult_, primaryPreProcessResultLen_).data());
 }
 
+void RequestProcessingState::detectNonDeterministicPreProcessing(const SHA3_256::Digest &newHash) const {
+  for (auto &hashArray : preProcessingResultHashes_)
+    if (newHash != hashArray.first) {
+      LOG_WARN(logger(),
+               "Received pre-processing result hash is different from calculated by other replica "
```

Comment:
It could be interesting to know which replica the non-matching hash came from

Reply:
Yeah, in case it is available.